inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasUimaParser.java

```diff
@@ -242,68 +242,158 @@ else if (annotation instanceof Sentence) {
+                                            endToken);
+                                    tokenCollection.add(mtasAnnotationTypeFeatureLabel);
+
+                                    indexedStr = annotationUiName.replace(" ", "_") + "."
+                                            + feature.getUiName().replace(" ", "_")
+                                            + MtasToken.DELIMITER + kbValue;
+
+                                    // Indexing UI annotation without type i.e Concept/Instance
+                                    log.debug("Indexed String without type for : {}", indexedStr);
+                                    MtasToken mtasAnnotationFeatureLabel = new MtasTokenString(
+                                            mtasId++, indexedStr, beginToken);
+                                    mtasAnnotationFeatureLabel.setOffset(annotation.getBegin(),
+                                            annotation.getEnd());
+                                    mtasAnnotationFeatureLabel.addPositionRange(beginToken,
+                                            endToken);
+                                    tokenCollection.add(mtasAnnotationFeatureLabel);
+                                    
+                                    // Indexing UI annotation without type and layer for generic search
+                                    indexedStr = IndexingConstants.kbEntity + MtasToken.DELIMITER
+                                            + kbValue;
+                                    log.debug("Indexed String without type and label for : {}", indexedStr);
+                                    MtasToken mtasAnnotationKBEntity = new MtasTokenString(
+                                            mtasId++, indexedStr, beginToken);
+                                    mtasAnnotationKBEntity.setOffset(annotation.getBegin(),
+                                            annotation.getEnd());
+                                    mtasAnnotationKBEntity.addPositionRange(beginToken,
+                                            endToken);
+                                    tokenCollection.add(mtasAnnotationKBEntity);
+                                    
+                                }
+
                             }
                         }
                     }
                 }
             }
-        }
-        catch (Exception e) {
+        } catch (Exception e) {
             log.error("Unable to index document", e);
         }
         return tokenCollection;
     }
-
+    
     @Override
-    public String printConfig()
-    {
+    public String printConfig() {
         return null;
     }
 
-    class SimpleAnnotationLayer
-    {
+    /**
+     * Takes in IRI for identifier and returns teh label String Eg: InputParameter :-
+     * http://www.w3.org/TR/2003/PR-owl-guide-20031209/wine#RoseDAnjou Returned :-
+     * "KBConcept"+MtasToken.DELIMITER+RoseDAnjou
```

Comment:
Use proper JavaDoc format markup, e.g. `{@code }`. Document parameters and return values using `@param` and `@return`, not in the javadoc body only.

Reply:
Done